process/txsimulator/interface.go

```diff
@@ -11,3 +12,8 @@ type TransactionProcessor interface {
 	VerifyTransaction(transaction *transaction.Transaction) error
 	IsInterfaceNil() bool
 }
+
+// DataFieldParser defines what a data field parser should be able to do
+type DataFieldParser interface {
+	Parse(dataField []byte, sender, receiver []byte, numOfShards uint32) *datafield.ResponseParseData
```

Comment:
IsInterfaceNil ?

Reply:
the data field parser from the `mx-chain-vm-common-go` does not implement this interface.